libkbfs/bare_root_metadata.go

```diff
@@ -29,3 +35,42 @@ func MakeInitialBareRootMetadata(
 
 	return MakeInitialBareRootMetadataV3(tlfID, h)
 }
+
+func dumpConfig() *spew.ConfigState {
+	c := spew.NewDefaultConfig()
+	c.Indent = "  "
+	c.DisablePointerAddresses = true
+	c.DisableCapacities = true
+	c.SortKeys = true
+	return c
+}
+
+// DumpBareRootMetadata returns a detailed dump of the given
+// BareRootMetadata's contents.
+func DumpBareRootMetadata(
+	codec kbfscodec.Codec, brmd BareRootMetadata) (string, error) {
+	serializedBRMD, err := codec.Encode(brmd)
+	if err != nil {
+		return "", err
+	}
+
+	// Make a copy so we can zero out SerializedPrivateMetadata.
+	brmdCopy, err := brmd.DeepCopy(codec)
+	if err != nil {
+		return "", err
+	}
+
+	switch brmdCopy := brmdCopy.(type) {
+	case *BareRootMetadataV2:
+		brmdCopy.SerializedPrivateMetadata = nil
+	case *BareRootMetadataV3:
+		brmdCopy.WriterMetadata.SerializedPrivateMetadata = nil
+	default:
+		// Do nothing, and let SerializedPrivateMetadata get
+		// spewed, I guess.
+	}
+	s := fmt.Sprintf("MD size: %d bytes\n"+
+		"MD version: %s\n\n", len(serializedBRMD), brmd.Version())
+	s += dumpConfig().Sdump(brmdCopy)
```

Comment:
I went the other way, actually. Using spew seems like an implementation detail to me, so I wanted to hide that. In the future if we want more control over the output we could expose it again, though.